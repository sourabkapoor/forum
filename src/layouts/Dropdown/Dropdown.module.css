.dropdown {
  position: absolute;
  /* top: 10px; */
  background-color: white;
  /* box-shadow: 0 1px 2px 0 rgba(255, 255, 255, 0.05); */
  box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
  /* outline: transparent solid 2px; */
  border-width: 1px;
  border-radius: 0.375rem;
  outline-offset: 2px;
  z-index: 1;
  margin: 5px 10px;
}

.community-width {
  width: 270px;
}

.profile-width {
  width: 176px;
}

.dropdown__content {
  max-height: 300px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 24px;
  padding-right: 24px;
  overflow-y: auto;
}

@media(max-width: 1210px) {
  .profile-width{
    width: fit-content;
    padding: 0;
    right: 20px;
  }
}
