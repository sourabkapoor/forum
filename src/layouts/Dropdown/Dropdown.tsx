import CSSModules from "react-css-modules";
import styles from "./Dropdown.module.css";
import { AiOutlinePlus } from "react-icons/ai";
import { useAppDispatch } from "../../hooks/hooks";
import { toggleCommunityModalState } from "../../features/subreddit/subredditSlice";

interface Props {
  children: JSX.Element | JSX.Element[];
  dropdown: string;
}
const Dropdown: React.FC<Props> = ({ children, dropdown }) => {
  return (
    <div
      styleName={`dropdown ${
        dropdown === "community" ? "community-width" : "profile-width"
      }`}
    >
      <div styleName="dropdown__content">{children}</div>
    </div>
  );
};

export default CSSModules(Dropdown, styles, {
  allowMultiple: true,
  handleNotFoundStyleName: "log",
});
