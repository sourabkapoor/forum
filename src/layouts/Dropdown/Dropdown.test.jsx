import { queryByPlaceholderText, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import each from "jest-each";
import Dropdown from "./Dropdown";

describe("Dropdown", () => {
  test("snapshot", () => {
    const { asFragment } = render(<Dropdown />);

    expect(asFragment()).toMatchSnapshot();
  });
});
