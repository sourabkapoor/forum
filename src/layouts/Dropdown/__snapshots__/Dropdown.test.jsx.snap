// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown snapshot 1`] = `
<DocumentFragment>
  <div
    class="dropdown profile-width"
  >
    <div
      class="dropdown__content"
    />
  </div>
</DocumentFragment>
`;
